### PR TITLE
Initial configuration

### LINKED ISSUE #82: To include the server stub and initial configuration
- [x] Generate server stub for the version 1.0.1 specification 
- [x] Copy the generated server stub to the server folder
- [x] Create a .gitignore file to exclude the server/node_modules 
- [x] Integrate load file to the server stub to the server/database folder

### LINKED ISSUE #83: To integrate BasicServices and ApplicationPattern to the server-side
- [x] In the packages.json include the following dependencies, 
"dependencies": {    
   "onf-core-model-ap": "2.0.2-alpha.1",
   "onf-core-model-ap-bs": "2.0.2-alpha.1"
  }

- [x] integrate BasicServices as per the steps in https://github.com/openBackhaul/ApplicationPattern/blob/develop/doc/ImplementingApplications/Steps2IntegrateBasicServices/Steps2IntegrateBasicServices.md
- [x] integrate security and logging handler

### LINKED ISSUE #84: To implement OAM Layer
The implementation of the following files shall be copy pasted from any of the TAC repositories.

controller/
- [x] Core.js
- [x] ActionProfile.js
- [x] GenericResponseProfile.js
- [x] HttpClient.js
- [x] HttpServer.js
- [x] IndividualServices.js
- [x] OperationClient.js
- [x] OperationServer.js
- [x] TcpClient.js
- [x] TcpServer.js

service/
- [x] CoreService.js
- [x] ActionProfileService.js
- [x] GenericResponseProfileService.js
- [x] HttpClientService.js
- [x] HttpServerService.js
- [x] IndividualServicesService.js
- [x] OperationClientService.js
- [x] OperationServerService.js
- [x] TcpClientService.js
- [x] TcpServerService.js

### PR DESCRIPTION
**Pull request by IswaryaaS**
Fixes #82
Fixes #83
Fixes #84

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// @ts-check
```
Comment (PrathibaJee) [inline server/index.js:1]: Very good. Ports are documented as expected.
All the initial functionalities are included.

File: server/controllers/Core.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var Core = require('../service/CoreService');
+var responseBuilder = require('onf-core-model-ap/applicationPattern/rest/server/ResponseBuilder');
+var responseCodeEnum = require('onf-core-model-ap/applicationPattern/rest/server/ResponseCode');
+var oamLogService = require('onf-core-model-ap/applicationPattern/services/OamLogService');
+
+module.exports.getControlConstruct = async function getControlConstruct(req, res, next) {
```
Comment (PrathibaJee) [inline server/controllers/Core.js:8]: While running the code, I got a 500 response code. I hope you have an issue for this. Please let me if you have one. Thanks. 
Sharing the error for our reference.
<details>
<summary>
Response Error
</summary>

{
  "code": 500,
  "message": ".response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0] should have required property 'operation-server-interface-1-0:operation-server-interface-pac', .response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0]['http-server-interface-1-0:http-server-interface-pac']['http-server-interface-capability']['data-update-period'] should be equal to one of the allowed values: http-server-interface-1-0:DATA_UPDATE_PERIOD_TYPE_REAL_TIME, http-server-interface-1-0:DATA_UPDATE_PERIOD_TYPE_1H_PERIOD, http-server-interface-1-0:DATA_UPDATE_PERIOD_TYPE_24H_PERIOD, http-server-interface-1-0:DATA_UPDATE_PERIOD_TYPE_MANUAL, .response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0] should have required property 'tcp-server-interface-1-0:tcp-server-interface-pac', .response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0] should have required property 'operation-client-interface-1-0:operation-client-interface-pac', .response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0] should have required property 'elasticsearch-client-interface-1-0:elasticsearch-client-interface-pac', .response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0] should have required property 'http-client-interface-1-0:http-client-interface-pac', .response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0] should have required property 'tcp-client-interface-1-0:tcp-client-interface-pac', .response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0] should match exactly one schema in oneOf",
  "errors": [
    {
      "path": ".response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0].operation-server-interface-1-0:operation-server-interface-pac",
      "message": "should have required property 'operation-server-interface-1-0:operation-server-interface-pac'",
      "errorCode": "required.openapi.validation"
    },
    {
      "path": ".response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0]['http-server-interface-1-0:http-server-interface-pac']['http-server-interface-capability']['data-update-period']",
      "message": "should be equal to one of the allowed values: http-server-interface-1-0:DATA_UPDATE_PERIOD_TYPE_REAL_TIME, http-server-interface-1-0:DATA_UPDATE_PERIOD_TYPE_1H_PERIOD, http-server-interface-1-0:DATA_UPDATE_PERIOD_TYPE_24H_PERIOD, http-server-interface-1-0:DATA_UPDATE_PERIOD_TYPE_MANUAL",
      "errorCode": "enum.openapi.validation"
    },
    {
      "path": ".response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0].tcp-server-interface-1-0:tcp-server-interface-pac",
      "message": "should have required property 'tcp-server-interface-1-0:tcp-server-interface-pac'",
      "errorCode": "required.openapi.validation"
    },
    {
      "path": ".response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0].operation-client-interface-1-0:operation-client-interface-pac",
      "message": "should have required property 'operation-client-interface-1-0:operation-client-interface-pac'",
      "errorCode": "required.openapi.validation"
    },
    {
      "path": ".response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0].elasticsearch-client-interface-1-0:elasticsearch-client-interface-pac",
      "message": "should have required property 'elasticsearch-client-interface-1-0:elasticsearch-client-interface-pac'",
      "errorCode": "required.openapi.validation"
    },
    {
      "path": ".response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0].http-client-interface-1-0:http-client-interface-pac",
      "message": "should have required property 'http-client-interface-1-0:http-client-interface-pac'",
      "errorCode": "required.openapi.validation"
    },
    {
      "path": ".response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0].tcp-client-interface-1-0:tcp-client-interface-pac",
      "message": "should have required property 'tcp-client-interface-1-0:tcp-client-interface-pac'",
      "errorCode": "required.openapi.validation"
    },
    {
      "path": ".response['core-model-1-4:control-construct']['logical-termination-point'][19]['layer-protocol'][0]",
      "message": "should match exactly one schema in oneOf",
      "errorCode": "oneOf.openapi.validation"
    }
  ]
}
</details>

Comment (IswaryaaS) [reply server/controllers/Core.js:8]: Yes Prathiba, we have documented few issues in ALAP regarding CC 500 error. Please find the list below:
https://github.com/openBackhaul/AutomatedLinkAcceptanceProxy/issues/107
https://github.com/openBackhaul/AutomatedLinkAcceptanceProxy/issues/106
https://github.com/openBackhaul/AutomatedLinkAcceptanceProxy/issues/108

**Review by PrathibaJee — APPROVED** [review]
Many thanks for fixing the review comments. 
Approving the changes.

**Pull request merged**